### PR TITLE
Migrate object utils

### PR DESCRIPTION
**Pull request by cesumilo**

### CODE REVIEW

File: mod.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+const {
+  DBError,
+  UniqueViolationError,
+  NotNullViolationError,
+  ForeignKeyViolationError,
+  ConstraintViolationError,
+  CheckViolationError,
+  DataError,
+} = require("db-errors");
+const { Model: NativeModel } = require("./model/Model");
```
Comment (yzalium) [inline mod.ts:12]: You're not importing using the full filename (with extension, ie `./model/Model.ts`)
I'm pretty sure Deno is now capable of handling extensionless imports, but I cannot find where for now 🤔

File: mod.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+const { ref } = require("./queryBuilder/ReferenceBuilder");
+const { val } = require("./queryBuilder/ValueBuilder");
+const { raw } = require("./queryBuilder/RawBuilder");
+const { fn } = require("./queryBuilder/FunctionBuilder");
+const { inherit } = require("../lib/utils/classUtils");
+
+// We need to wrap the classes, that people can inherit, with ES5 classes
+// so that babel is able to use ES5 inheritance. sigh... Maybe people
+// should stop transpiling node apps to ES5 in the year 2019? Node 6
+// with full class support was released three years ago.
```
Comment (yzalium) [inline mod.ts:55]: Maybe this issue is irrelevant now in Deno ?

File: mod.ts (hunk 1 above)
Comment (cesumilo) [reply mod.ts:12]: Ah sorry.. I didn't migrate anything in mod.ts 😆 I just renamed and that's it. The only part I've been migrating is src/utils/object.ts

File: mod.ts (hunk 2 above)
Comment (cesumilo) [reply mod.ts:55]: Same here: I just renamed and that's it. The only part I've been migrating is src/utils/object.ts

**Review by yzalium — COMMENTED** [review]
Just have two questions, in the comments below
Also, are we planning to rewrite the js along the way ?

**Comment by cesumilo** [comment]
> Just have two questions, in the comments below Also, are we planning to rewrite the js along the way ?

Only relevant part of it. No backward compatibility imo?

File: src/utils/object.ts [hunk 3]
```diff
@@ -0,0 +1,542 @@
...
+    const value = arrays[i];
+
+    if (Array.isArray(value)) {
+      for (let j = 0; j < value.length; ++j) {
+        out.push(value[j]);
+      }
+    } else {
+      out.push(value);
+    }
+  }
```
Comment (yzalium) [inline src/utils/object.ts:227]: Can't we just replace it with 
```suggestion
  return arrays.flat(Infinity);
```

**Review by yzalium — APPROVED** [review]

**Pull request merged**